src/main/java/com/artipie/cargo/meta/MetadataMerge.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/cargo-adapter/LICENSE
+ */
+package com.artipie.cargo.meta;
+
+import com.artipie.asto.ArtipieIOException;
+import com.artipie.asto.misc.UncheckedIOConsumer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+/**
+ * Merge new package metadata into optionally existing metadata file.
+ * @since 0.1
+ */
+public final class MetadataMerge implements BiConsumer<Optional<InputStream>, OutputStream> {
```

Comment:
@olenagerasimova thanks, one question. Is it necessary to eliminate duplicate lines? Or it is impossible because each line has some unique information

Reply:
@genryxy duplicates will be checked on upper level, before merging metadata. So here duplicates are not possible